lib/serialization/dictionarySpec.ts

```diff
@@ -60,9 +72,21 @@ export function dictionarySpec<TSerializedValue, TDeserializedValue>(valueSpec:
 
         result = value as any;
       } else {
+        let valueTypeSpec: TypeSpec<TSerializedValue, TDeserializedValue>;
+        if (typeof valueSpec === "string") {
+          if (!options.compositeSpecDictionary || !options.compositeSpecDictionary[valueSpec]) {
+            const errorMessage = `Missing composite specification entry in composite type dictionary for type named "${valueSpec}" at property ${propertyPath}.`;
+            log(options, HttpPipelineLogLevel.ERROR, errorMessage);
+            throw new Error(errorMessage);
+          }
+          valueTypeSpec = options.compositeSpecDictionary[valueSpec] as TypeSpec<TSerializedValue, TDeserializedValue>;
+        } else {
+          valueTypeSpec = valueSpec;
+        }
```

Comment:
Since serialize and deserialize have the same code repeated twice for dictionary and for sequence and I am sure it would be the same for a CompositeType referencing another CompositeType, how about making this a function and then calling that function in different places. This avoids code duplication and reduces copy paste errors.